tutorials/navigation/navigation_using_navigationmeshes.rst

```diff
@@ -17,254 +19,333 @@ Navigation meshes are the only things considered when doing pathfinding, e.g. vi
+The geometry data should be in general kept very simple. As many edges as are required but as few as possible.
+Especially in 2D duplicated and nested geometry should be avoided as it forces polygon hole calculation that can result in flipped polygons.
+An example for nested geometry would be a smaller StaticBody2D shape placed completely inside the bounds of another StaticBody2D shape.
 
-    var new_navigation_polygon: NavigationPolygon = NavigationPolygon.new()
-    var new_outline: PackedVector2Array = PackedVector2Array([
-        Vector2(0.0, 0.0),
-        Vector2(50.0, 0.0),
-        Vector2(50.0, 50.0),
-        Vector2(0.0, 50.0),
-    ])
-    new_navigation_polygon.add_outline(new_outline)
-    new_navigation_polygon.make_polygons_from_outlines()
+Navigation mesh baking common problems
+--------------------------------------
 
-    NavigationServer2D.region_set_navigation_polygon(new_2d_region_rid, new_navigation_polygon)
+There are some common user problems and important caveats to consider when creating or baking navigation meshes.
 
-Procedural 3D navigation mesh
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+- Navigation mesh baking creates frame rate problems at runtime
+    The navigation mesh baking is by default done on a background thread so as long as the platform supports threads the actual baking is
+    rarely the source of any performance issues (assuming a reasonably sized and complex geometry for runtime rebakes).
 
-The following script creates a new 3D navigation region and fills it with procedurally generated navigation mesh data from a NavigationMesh resource.
+    The common source for performance issues at runtime is the parsing step for source geometry that involves nodes and the SceneTree.
+    The SceneTree is not thread-safe so all the nodes need to be parsed on the main-thread.
+    Some nodes with a lot of data can be very heavy and slow to parse at runtime, e.g. a TileMap has one or more polygons for every single used cell and TileMapLayer to parse.
+    Nodes that hold meshes need to request the data from the RenderingServer stalling the rendering in the process.
+
+    To improve performance use more optimized shapes, e.g. collision shapes over detailed visual meshes, and merge and simplify as much geometry as possible upfront.
+    If nothing helps don't parse the SceneTree and add the source geometry procedural with scripts. If only pure data arrays are used as source geometry the entire baking process can be done on a background thread.
+
+- Navigation mesh creates unintended holes in 2D.
+    The navigation mesh baking in 2D is done by doing polygon clipping operations based on outline paths.
+    Polygons with "holes" are a necessary evil to create more complex 2D polygons but can become unpredictable for users with many complex shapes involved.
+
+    To avoid any unexpected problems with polygon hole calculations avoid nesting any outlines inside other outlines for the same type (traversable / obstruction).
```

Comment:
```suggestion
    To avoid any unexpected problems with polygon hole calculations, avoid nesting any outlines inside other outlines of the same type (traversable / obstruction).
```